agent/agent_endpoint.go

```diff
@@ -344,7 +344,8 @@ func (s *HTTPServer) AgentRegisterCheck(resp http.ResponseWriter, req *http.Requ
 
 	// Verify the check type.
 	chkType := args.CheckType()
-	if !chkType.Valid() {
+	_, err := chkType.Valid()
+	if err != nil {
 		resp.WriteHeader(http.StatusBadRequest)
 		fmt.Fprint(resp, invalidCheckMessage)
```

Comment:
Now that we have a more detailed error, what do you think about adding the error along with the feedback (here and on line 576)?